### ISSUE TITLE
torch.jit.script is swallowing a rfloordiv warning

### ISSUE THREAD

**Issue by mattip** [pytorch/pytorch#52741]
## 🐛 Bug

Wrapping a rfloordiv operation with `torch.jit.script` somehow swallows the warning:

## To Reproduce
```
# Run with python -Walways
import torch

def wrap_div(a):
    return 5 // a

script_wrap_div = torch.jit.script(wrap_div)

a = torch.tensor(5, dtype=int)

print('This does not warn')
script_wrap_div(a)

print('This warns')
wrap_div(a)
```

will print
```
$ python -Walways /tmp/stam.py
This does not warn
This warns
pytorch/torch/tensor.py:563: UserWarning: floor_divide is deprecated, and will be removed in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values.
To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor'). (Triggered internally at  ../aten/src/ATen/native/BinaryOps.cpp:335.)
  return torch.floor_divide(other, self)
```
## Expected behavior
Between the `This does not warn` and the `This warns` lines, I expected to see the warning

## Environment

PyTorch version: 1.9.0a0+8b0ca1a
Is debug build: True
CUDA used to build PyTorch: 11.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (x86_64)
GCC version: (crosstool-NG 1.24.0.133_b0863d8_dirty) 9.3.0
Clang version: 11.0.1 (https://github.com/conda-forge/clangdev-feedstock 9fbb64b62ff49e9d206a06e62453b27557b3ed73)
CMake version: version 3.19.4

Python version: 3.7 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 11.2.142
GPU models and configuration: GPU 0: GeForce GTX 1660 SUPER
Nvidia driver version: 460.32.03
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip] numpy==1.20.0
[pip] pytorch-sphinx-theme==0.0.24
[pip] torch==1.9.0a0+8b0ca1a
[conda] cudatoolkit               11.2.0               h73cb219_7    conda-forge
[conda] magma-cuda112             2.5.2                         1    pytorch
[conda] mkl                       2020.4             h726a3e6_304    conda-forge
[conda] mkl-include               2020.4             h726a3e6_304    conda-forge
[conda] numpy                     1.20.0           py37haa41c4c_0    conda-forge
[conda] pytorch-sphinx-theme      0.0.24                    dev_0    <develop>
[conda] torch                     1.9.0a0+8b0ca1a           dev_0    <develop>

## Additional context

Discovered when working on PR gh-52387, part of issue gh-50006

**Comment by peterbell10** [comment]
The graph contains `aten::floordiv` instead of `aten::floor_div` which seems to have different behavior. e.g.
```python
>>> a = torch.randint(1, 5, size=(5,))
>>> wrap_div(a)
tensor([5, 1, 1, 1, 1])

>>> script_wrap_div(a)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-17-2742771fd276> in <module>
----> 1 script_wrap_div(a)

RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
  File "<ipython-input-14-97566baba668>", line 4, in wrap_div
def wrap_div(a: torch.Tensor):
    return 5 // a
           ~~~~~~ <--- HERE
RuntimeError: Cannot input a tensor of dimension other than 0 as a scalar argument
```

**Comment by Lilyjjo** [comment]
Notes: we need to change JIT to use the same operator as eager

**Comment by nikithamalgifb** [comment]
Looks, like we automatically convert the tensor to an int here:

Here is the graph:
```
graph(%a.1 : Tensor):
  %7 : bool = prim::Constant[value=0]()
  %6 : None = prim::Constant()
  %5 : int = prim::Constant[value=3]() # repro.py:142:39
  %2 : str = prim::Constant[value="A:::"]() # repro.py:141:10
  %1 : str = prim::Constant[value="Inside wrap_div"]() # repro.py:140:10
  %4 : int = prim::Constant[value=5]() # repro.py:142:11
   = prim::Print(%1) # repro.py:140:4
   = prim::Print(%2, %a.1) # repro.py:141:4
  %8 : Tensor = aten::tensor(%4, %5, %6, %7) # repro.py:142:17
  %9 : int = aten::IntImplicit(%8) # repro.py:142:11
  %10 : int = aten::floor_divide(%4, %9) # repro.py:142:11
  return (%10)
```
For JIT:
=======
A is 5
Output : 1

For Eager:
=========
A is tensor(5)
/data/users/nikithamalgi/pytorch/torch/tensor.py:569: UserWarning: floor_divide is deprecated, and will be removed in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values.
To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor'). (Triggered internally at  ../aten/src/ATen/native/BinaryOps.cpp:341.)
  return torch.floor_divide(other, self)
Output: tensor(1, dtype=torch.int32)

The return value of a scripted wrap_div,  is an integer while Eager mode returns a tensor. 

@gmagogsfm , @eellison : Since 5 is an int, we implicitly convert the tensor to an int. Is this expected behavior?

**Comment by gmagogsfm** [comment]
> Looks, like we automatically convert the tensor to an int here:
> 
> Here is the graph:
> 
> ```
> graph(%a.1 : Tensor):
>   %7 : bool = prim::Constant[value=0]()
>   %6 : None = prim::Constant()
>   %5 : int = prim::Constant[value=3]() # repro.py:142:39
>   %2 : str = prim::Constant[value="A:::"]() # repro.py:141:10
>   %1 : str = prim::Constant[value="Inside wrap_div"]() # repro.py:140:10
>   %4 : int = prim::Constant[value=5]() # repro.py:142:11
>    = prim::Print(%1) # repro.py:140:4
>    = prim::Print(%2, %a.1) # repro.py:141:4
>   %8 : Tensor = aten::tensor(%4, %5, %6, %7) # repro.py:142:17
>   %9 : int = aten::IntImplicit(%8) # repro.py:142:11
>   %10 : int = aten::floor_divide(%4, %9) # repro.py:142:11
>   return (%10)
> ```
> 
> ## For JIT:
> A is 5
> Output : 1
> 
> ## For Eager:
> A is tensor(5)
> /data/users/nikithamalgi/pytorch/torch/tensor.py:569: UserWarning: floor_divide is deprecated, and will be removed in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values.
> To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor'). (Triggered internally at ../aten/src/ATen/native/BinaryOps.cpp:341.)
> return torch.floor_divide(other, self)
> Output: tensor(1, dtype=torch.int32)
> 
> The return value of a scripted wrap_div, is an integer while Eager mode returns a tensor.
> 
> @gmagogsfm , @eellison : Since 5 is an int, we implicitly convert the tensor to an int. Is this expected behavior?

Sorry for not having enough context, do you mean TS compiles into `aten::floor_divide` because of implicit conversion from `tensor` to `int`? Had such conversion not happen, TS would exhibit same behavior as eager?

**Comment by nikithamalgifb** [comment]
Yeah, that's right. In JIT,if we don't convert the `tensor` to `int`, we would basically call aten::floor_divide(int, tensor) which returns a `tensor` instead of aten::floor_divide(int, int) which returns a `int` value.

**Comment by Apurv-Bhardwaj** [comment]
Is this issue still up?

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
